### PR TITLE
chore(security-issues): fixed springboot security vulnerabilities and added codeql

### PR DESCRIPTION
**Pull request by saudkhan116**
# Why we create this PR?
 
- Added CodeQL workflow to analyze the codebase as a requirement from the release guidelines
- Removed veracode workflows from git actions (replaced by CodeQL).
- Fixed spring security vulnerabilities by upgrading the springboot version from `3.2.2` to `3.2.4`

 
# What we want to achieve with this PR?
 
To meet the [trg-8](https://eclipse-tractusx.github.io/docs/release/trg-0/trg-8-05) requirements

 
# What is new?
 
## Added
- Added CodeQL code analysis to the workflow to comply with the release guidelines

## Deleted
- Removed veracode workflows from git actions (replaced by CodeQL).

## Security Fixes:
- Fixed the following spring security vulnerabilities:
   - CVE-2024-22234
   -  CVE-2024-22257
   - CVE-2024-22243
   - CVE-2024-22259

Linked PR(s):

#289

### CODE REVIEW

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        # Use only 'java' to analyze code written in Java, Kotlin or both
+        # Use only 'javascript' to analyze code written in JavaScript, TypeScript or both
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup JDK 21
+        uses: ./.github/actions/setup-java
```
[the remark this answers is not in the document]
Comment (RoKrish14) [reply .github/workflows/codeql.yaml:66]: Similar to previous alert:
Consider using the latest release checksum value to validate the SHA alerts:

uses: actions/setup-java@99b8673ff64fbf99d8d325f52d9a5bdedb8483e9 # v4.2.1

**Review by RoKrish14 — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (saudkhan116) [reply .github/workflows/codeql.yaml:66]: the issue is now fixed

**Review by matbmoser — APPROVED** [review]
If all dependencies are approved and the workflow changes propose by @RoKrish14 are solved then I approve

**Comment by matbmoser** [comment]
@RoKrish14  please check and re approve

**Comment by saudkhan116** [comment]
> If all dependencies are approved and the workflow changes propose by @RoKrish14 are solved then I approve

@matbmoser, I fixed now the issues requested by @RoKrish14. I hope there should be no issue left behind.

**Review by RoKrish14 — APPROVED** [review]
LGTM 👍

File: .github/actions/setup-java/action.yaml [hunk 2]
```diff
@@ -28,7 +28,7 @@ runs:
   using: "composite"
   steps:
     - name: Setup Java
-      uses: actions/setup-java@v4
+      uses: actions/setup-java@99b8673ff64fbf99d8d325f52d9a5bdedb8483e9   # v4.2.1
```
Comment (saudkhan116) [inline .github/actions/setup-java/action.yaml:31]: @RoKrish14 and @matbmoser, the issue is fixed here.

**Pull request merged**